rdmd.d

```diff
@@ -916,3 +891,71 @@ void yap(size_t line = __LINE__, T...)(auto ref T stuff)
     debug stderr.writeln(line, ": ", stuff);
     else stderr.writeln(stuff);
 }
+
+/**
+Used to wrap filesystem operations that should also be logged with the
+yap function or affected by dryRun. Append the string "IfLive" to the end
+of the function for it to be skipped during a dry run.
+
+These functions allow the filename to be given once so the log statement
+always matches the operation. Using it also guarantees you won't forget to
+include a `yap` alongside any file operation you want to be logged.
+*/
+struct Filesystem
+{
+static:
+    static auto opDispatch(string func, T...)(T args)
```

Comment:
`static` on function is redundant

Reply:
I think the redundancy is ok....if this struct had really big functions, and you couldn't see the `static:` modifier, having `static` in the function definition is helpful.  But if you really don't like the redundancy, I'll remove it.

Reply:
Ah OK, I didn't realize it was intentional.